processing/src/main/java/io/druid/query/aggregation/AggregatorFactory.java

```diff
@@ -32,13 +32,13 @@
  * provided to the Aggregator through the MetricSelector object, so whatever creates that object gets to choose how
  * the data is actually stored and accessed.
  */
-public interface AggregatorFactory
+public abstract class AggregatorFactory
```

Comment:
The primary benefit of switching to an abstract class is that it allows us to add new methods to the interface without necessarily causing a backwards incompatible change.  That is, we should consider this an "interface" that just happens to have some default implementations that allowed us to add the methods without a backwards incompatible change.

This is only true if we can do a default implementation that is meaningful for all implementations though.  If we are adding a method that won't have a good default implementation we believe is true for everyone, then we should make it a backwards incompatibility and force anyone who has implemented this interface to think about what is required to actually implement it correctly.